docs/source/blogs/tech_blog/blog12_Inference_Time_Compute_Implementation_in_TensorRT-LLM.md

```diff
@@ -0,0 +1,425 @@
+# Inference Time Compute Implementation in TensorRT LLM
+
+By NVIDIA TensorRT LLM Team
+
+## Table of Contents
+- [Inference-Time Compute Implementation in TensorRT LLM (Part 1: Design and Implementation](#inference-time-compute-implementation-in-tensorrt-llm)
+  - [Table of Content](#table-of-content)
+  - [Background and Motivation](#background-and-motivation)
+  - [Introduction for Scaffolding: A Framework for inference-time compute](#introduction-for-scaffolding)
+    - [Core Features](#scaffolding-core-feature)
+    - [Architecture](#scaffolding-architecture)
+      - [Worker](#scaffolding-architecture-worker)
+      - [Controller](#scaffolding-architecture-controller)
+      - [ScaffoldingLlm](#scaffolding-architecture-scaffoldingllm)
+  - [An Example: Implement Dynasor on Scaffolding](#example-for-scaffolding)
+    - [Introduction for Dynasor](#dynasor-introduction)
+    - [Implement Dynasor-CoT in Scaffolding](#dynasor-cot-implement-in-scaffolding)
+    - [Implement Dynasor-CoT based Majority Voting in Scaffolding](#dynasor-cot-based-majority-vote-in-scaffolding)
+  - [Feature List on Scaffolding](#scaffolding-feature-list)
+  - [Future Work](#scaffolding-future-work)
+
+
+## Background and Motivation
+Inference-time compute (aka test-time scaling) is becoming increasingly important. In addition to simply increasing the length of the output, using various workflows such as best-of-N and MCTS (Monte Carlo Tree Search) to obtain better answers is also an important means. Further, most of the workflows of agentic or multi-agent are logically similar to these methods of inference-time compute, except that they use more complex tools and context engineering. However, how to conveniently define these methods while achieving excellent inference performance has become a new problem. Because good performance requires careful asynchronous scheduling, but writing asynchronous scheduling programs is not easy for algorithm engineers. When considering the use of external tools and token budget management, the problem becomes even more complex.
```

Comment:
suggestion:
"Inference-time compute (aka test-time scaling) is becoming increasingly important. In addition to simply increasing the length of the output, using various workflows such as best-of-N and MCTS (Monte Carlo Tree Search) to obtain better answers is also an important means."
->
"Inference-time compute, also known as test-time scaling, is increasingly important. Beyond simply increasing output length, workflows such as best-of-N and Monte Carlo Tree Search (MCTS) offer additional capabilities for optimizing inference."